subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/README.md

```diff
@@ -1,24 +1,43 @@
+### Topology-Aware Resource Guarantee
 
-Lastly, HiveD is **fault-tolerant**. It can handle random machine failures and crash recovery.
+The killer feature that distinguishes HiveD is that it provides resource guarantee to each VC, not only in terms of quantity, a numeric value, but also in terms of **[topology](example/feature/README.md#VC-Safety)**, a key requirement of GPU-based training jobs. For example, a traditional scheduler guarantees that a VC can use 8 GPUs. However, it does not know the topology of these 8 GPUs. It is possible that an 8-GPU training job which has to run within a single node, cannot be allocated even if its VC still has 8 free GPUs. This is because these 8 free GPUs may belong to multiple nodes in the physical cluster.
+
+HiveD protects VCs' resources in terms of **cell**, a user-defined resource type that encodes both the quantity and the topology of the GPUs (among other specific hardware configurations, such as GPU type, networking). In the above example, a user can define a cell type of 8-GPU node, and the VC can be assigned one of such cell. Then, HiveD will ensure that *there is always one 8-GPU node available for the VC*, regardless of the other workloads in the cluster.
+
+HiveD allows flexible cell definitions and assignments. For example, users can define cells at multiple topology levels (e.g., PCI-e switch) for finer-grained allocations. Users can also create different cell types for different GPU models (e.g., V100) or networking configuration (e.g., InfiniBand domain) in a heterogeneous cluster.
+
+### Gang Scheduling
+
+HiveD optimizes the performance of **[gang scheduling](example/feature/README.md#Gang-Scheduling)**, a typical scheduling requirement for deep learning training jobs, where all containers should be allocated before the training job can begin. Multiple gang-scheduled jobs competing for the same set of resource may lead to starvation, where each job only gets partial resource and has to wait indefinitely.
+
+HiveD schedules all containers within a job in a *transactional* manner, i.e., all these containers' requirements will be granted or denied as a whole, thus avoiding partial resource allocation and starvation.
+
+### Priorities
+
+HiveD supports multiple job **priorities**. Higher-priority jobs can **[preempt](example/feature/README.md#Intra-VC-Preemption)** lower-priority jobs. HiveD also introduces **[opportunistic jobs](example/feature/README.md#Opportunistic-Job)**, i.e., jobs with the lowest priority which can use other VCs' free resource when possible (without breaking the resource guarantees to other VCs).
+
+### Fault-Tolerance
+
+HiveD is **fault-tolerant**. It can handle random failures of both the scheduler itself and the resource it manages.
```

Comment:
>and the resource it manages. [](start = 88, length = 28)

fault-tolerant just mean hived itself is fault-tolerant. Bad HW Awareness is another feature. I think Fault-Tolerance is a general common feature, so let's remove it from these key features?